pandas/core/algorithms.py

```diff
@@ -72,6 +72,7 @@
     na_value_for_dtype,
 )
 
+import pandas as pd
```

Comment:
we try to avoid importing this internally.  pls use a more specific import

Reply:
Done. Put it in the function since it circular imports outside of it.